src/libponyc/pass/docgen.c

```diff
@@ -483,7 +500,22 @@ static void doc_type_list(docgen_t* docgen, docgen_opt_t* docgen_opt, ast_t* lis
   fprintf(docgen->type_file, "%s", postamble);
 }
 
-
+static void add_source_code_link(docgen_t* docgen, ast_t* elem)
+{
+  source_t* source = ast_source(elem);
+  const doc_sources_t* doc_source = NULL;
+
+  if (source != NULL)
+    doc_source = get_doc_source(docgen, source);
+
+  if (doc_source != NULL) {
+      fprintf(
+        docgen->type_file, 
+        "<div class=\"source-link\">[[Source]](%s#L%zd)</div>", 
```

Comment:
please put this div on a new line, otherwise it appears as part of the heading. i will style it to go right after the heading (possibly on the same line if there is enough space).

Reply:
Done.